src/recipes/ai-context/index.ts

```diff
@@ -48,78 +70,171 @@ const promptForPath = async (): Promise<string> => {
   ])
 
   if (customPath) {
-    return customPath
+    return { ...rulesForDefaultConsumer, path: customPath || rulesForDefaultConsumer.path }
   }
 
   log('You must select a path.')
 
-  return promptForPath()
+  return promptForContextConsumerSelection()
+}
+
+/**
+ * Checks if a command belongs to a known IDEs by checking if it includes a specific string.
+ * For example, the command that starts windsurf looks something like "/applications/windsurf.app/contents/...".
+ */
+const getConsumerKeyFromCommand = (command: string): string | null => {
+  // The actual command is something like "/applications/windsurf.app/contents/...", but we are only looking for windsurf
+  const match = cliContextConsumers.find(
+    (consumer) => consumer.consumerProcessCmd && command.includes(consumer.consumerProcessCmd),
+  )
+  return match ? match.key : null
+}
+
+/**
+ * Receives a process ID (pid) and returns both the command that the process was run with and its parent process ID. If the process is a known IDE, also returns information about that IDE.
+ */
+const getCommandAndParentPID = async (
+  pid: number,
+): Promise<{
+  parentPID: number
+  command: string
+  consumerKey: string | null
+}> => {
+  const { stdout } = await execa('ps', ['-p', String(pid), '-o', 'ppid=,comm='])
+  const output = stdout.trim()
+  const spaceIndex = output.indexOf(' ')
+  const parentPID = output.substring(0, spaceIndex)
+  const command = output.substring(spaceIndex + 1).toLowerCase()
+  return {
+    parentPID: parseInt(parentPID, 10),
```

Comment:
nit: This should be faster and a bit nicer.

```suggestion
    parentPID: Number(parentPID),
```